File: php/features.php
<?php
declare(strict_types=1);

// MercadoBitcoin SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MercadoBitcoinFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MercadoBitcoinBaseFeature();
            case "test":
                return new MercadoBitcoinTestFeature();
            default:
                return new MercadoBitcoinBaseFeature();
        }
    }
}

File: py/features.py
# MercadoBitcoin SDK feature factory

from feature.base_feature import MercadoBitcoinBaseFeature
from feature.test_feature import MercadoBitcoinTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MercadoBitcoinBaseFeature(),
        "test": lambda: MercadoBitcoinTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
